src/mapblock.h

```diff
@@ -642,23 +642,23 @@ inline bool objectpos_over_limit(v3f p)
 	const static float map_gen_limit_bs = MYMIN(MAX_MAP_GENERATION_LIMIT,
 		g_settings->getU16("map_generation_limit")) * BS;
 	return (p.X < -map_gen_limit_bs
-		|| p.X >  map_gen_limit_bs
+		|| p.X > map_gen_limit_bs
 		|| p.Y < -map_gen_limit_bs
-		|| p.Y >  map_gen_limit_bs
+		|| p.Y > map_gen_limit_bs
 		|| p.Z < -map_gen_limit_bs
-		|| p.Z >  map_gen_limit_bs);
+		|| p.Z > map_gen_limit_bs);
 }
 
 inline bool blockpos_over_limit(v3s16 p)
 {
 	const static u16 map_gen_limit = MYMIN(MAX_MAP_GENERATION_LIMIT,
 		g_settings->getU16("map_generation_limit"));
-	return (p.X < -map_gen_limit / MAP_BLOCKSIZE
-			|| p.X >  map_gen_limit / MAP_BLOCKSIZE
-			|| p.Y < -map_gen_limit / MAP_BLOCKSIZE
-			|| p.Y >  map_gen_limit / MAP_BLOCKSIZE
-			|| p.Z < -map_gen_limit / MAP_BLOCKSIZE
-			|| p.Z >  map_gen_limit / MAP_BLOCKSIZE);
+	return (p.X * MAP_BLOCKSIZE < -map_gen_limit 
+		|| (p.X + 1) * MAP_BLOCKSIZE - 1 > map_gen_limit
```

Comment:
The min nodes of blocks are at (n * mapblocksize)
So the max nodes are at (n * mapblocksize) - 1